From 0b04fa5aaca85b9e73a7d94bb5499e62df7bf025 Mon Sep 17 00:00:00 2001
From: David Edmundson <davidedmundson@kde.org>
Date: Mon, 29 Jun 2026 23:11:51 +0100
Subject: [PATCH] wayland: Drop outdated guard for empty windows

A guard was added in 2014 (439a084cd359acfafa284961b5f08d6b2891c6c6 in
QtWayland) to guard against calling
wl_egl_window_create/wl_egl_window_resize with an empty size.

A recent refactor moved (ed4bf9c993041b0cbb2ceb07049bb928469b66e8) it
from changing the window lifespan to being in makeCurrent. This broke
code required an opengl context but didn't show a window.

It doesn't make sense here, we've already called wl_egl_window_create
with the size by this point.

Pick-to: 6.12
Change-Id: I7721a8cf6b6abe754975f8bfbaffceda0ec05492
Reviewed-by: David Redondo <qt@david-redondo.de>
---
 .../wayland-egl/qwaylandglcontext.cpp                    | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/src/plugins/platforms/wayland/plugins/hardwareintegration/wayland-egl/qwaylandglcontext.cpp b/src/plugins/platforms/wayland/plugins/hardwareintegration/wayland-egl/qwaylandglcontext.cpp
index e8d3c2c2672..0a155a2fbe0 100644
--- a/src/plugins/platforms/wayland/plugins/hardwareintegration/wayland-egl/qwaylandglcontext.cpp
+++ b/src/plugins/platforms/wayland/plugins/hardwareintegration/wayland-egl/qwaylandglcontext.cpp
@@ -436,15 +436,6 @@ bool QWaylandGLContext::makeCurrent(QPlatformSurface *surface)
     EGLSurface eglSurface = m_currentWindow->eglSurface();
     QSize size = m_currentWindow->bufferSize();
 
-    // wl_egl_windows must have both width and height > 0
-    // mesa's egl returns NULL if we try to create a, invalid wl_egl_window, however not all EGL
-    // implementations may do that, so check the size ourself. Besides, we must deal with resizing
-    // a valid window to 0x0, which would make it invalid. Hence, destroy it.
-    if (size.isEmpty()) {
-        lock.unlock();
-        m_currentWindow->invalidateSurface();
-        return true;
-    }
     if (eglSurface == EGL_NO_SURFACE) {
         m_currentWindow->createSurface(size);
         eglSurface = m_currentWindow->eglSurface();
-- 
GitLab

